src/utils/hooks/transaction/utils/submit.ts

```diff
@@ -0,0 +1,106 @@
+import { ExtrinsicData } from '@interlay/interbtc-api';
+import { ApiPromise } from '@polkadot/api';
+import { AddressOrPair, SubmittableExtrinsic } from '@polkadot/api/types';
+import { DispatchError } from '@polkadot/types/interfaces';
+import { ExtrinsicStatus } from '@polkadot/types/interfaces/author';
+import { ISubmittableResult } from '@polkadot/types/types';
+
+import { TransactionEvents } from '../types';
+
+type HandleTransactionResult = { result: ISubmittableResult; unsubscribe: () => void };
+
+// When passing { nonce: -1 } to signAndSend the API will use system.accountNextIndex to determine the nonce
+const transactionOptions = { nonce: -1 };
+
+const handleTransaction = async (
+  account: AddressOrPair,
+  extrinsicData: ExtrinsicData,
+  expectedStatus?: ExtrinsicStatus['type'],
+  callbacks?: TransactionEvents
+) => {
+  let isComplete = false;
+
+  // Extrinsic status
+  let isReady = false;
+
+  return new Promise<HandleTransactionResult>((resolve, reject) => {
+    let unsubscribe: () => void;
+
+    (extrinsicData.extrinsic as SubmittableExtrinsic<'promise'>)
+      .signAndSend(account, transactionOptions, callback)
+      .then((unsub) => (unsubscribe = unsub))
+      .catch((error) => reject(error));
+
+    function callback(result: ISubmittableResult): void {
+      const { onReady } = callbacks || {};
+
+      if (!isReady && result.status.isReady) {
+        onReady?.();
+        isReady = true;
+      }
+
+      if (!isComplete) {
+        isComplete = expectedStatus === result.status.type;
+      }
+
+      if (isComplete) {
+        resolve({ unsubscribe, result });
+      }
+    }
+  });
+};
+
+const getErrorMessage = (api: ApiPromise, dispatchError: DispatchError) => {
+  const { isModule, asModule, isBadOrigin } = dispatchError;
+
+  // Construct error message
+  const message = 'The transaction failed.';
+
+  // Runtime error in one of the parachain modules
+  if (isModule) {
+    // for module errors, we have the section indexed, lookup
+    const decoded = api.registry.findMetaError(asModule);
+    const { docs, name, section } = decoded;
+    return message.concat(` The error code is ${section}.${name}. ${docs.join(' ')}`);
+  }
+
+  // Bad origin
+  if (isBadOrigin) {
+    return message.concat(` The error is caused by using an incorrect account.
+        The error code is BadOrigin ${dispatchError}.`);
```

Comment:
Won't there be too much whitespace in error message?
```suggestion
    return message.concat(` The error is caused by using an incorrect account. The error code is BadOrigin${dispatchError}.`);
```

Reply:
This was copied from the lib. So, im unsure.